File: missing-numbers/missing-numbers.py
#hackerRank - Missing numbers

#1

def missingNumbers(arr, brr):
    # Write your code here
    b_counter = {}
    for num_b in brr:
        if num_b in b_counter:
            b_counter[num_b] += 1
        else:
            b_counter[num_b] = 1
    result = []
    for num_a in arr:
        if num_a in b_counter:
            b_counter[num_a] -= 1
        else:
            result.append(num_a)
    for key, value in b_counter.items():
        if value > 0:
            result.append(key)
    return sorted(result)

File: missing-numbers/missing-numbers.js
//hackerRank - Missing numbers

function missingNumbers(arr, brr) {
    // Write your code here
    const bCounter = new Map();
    for (let i = 0; i < brr.length; i++) {
        if (bCounter.has(brr[i])) {
            bCounter.set(brr[i], bCounter.get(brr[i]) + 1)
        } else {
            bCounter.set(brr[i], 1)
        }
    };
    console.log(bCounter);
    let result = [];
    for (let j = 0; j < arr.length; j++) {
        if (bCounter.has(arr[j])) {
            bCounter.set(arr[j], bCounter.get(arr[j]) - 1)
        } else {
            result.push(arr[j])
        };
    };
    console.log(bCounter);
    for (const [key, value] of bCounter.entries()) {
        if (value > 0) {
            result.push(key)
        }
    };
    return result.sort((a, b) => a - b)
}
